### PR TITLE
ui-ux: debug tab

### PR DESCRIPTION
**Pull request by dustinwloring1988**
On Debug tab:
Fixed check for update to get the correct commit for the lastest hash not the hash from the commit.json in the main branch as the latest is after the action that adds the commit
Fixed 'Error: No URL configured' error in local llm status section when entered

### CODE REVIEW

**Comment by dustinwloring1988** [comment]
@thecodacus can you please test this on your setup it work on mine.

**Comment by thecodacus** [comment]
looks great. can we have updated from stable branch and main branch ?

**Comment by dustinwloring1988** [comment]
@thecodacus Yes great idea maybe stable branch by default and a toggle for ('latest' or 'dev')? Should that be a separate PR though, if so I will work on it today as well.

**Comment by dustinwloring1988** [comment]
or do we want to put in the commit file which branch it is in so it only updates from that one?

**Comment by dustinwloring1988** [comment]
image

**Comment by thecodacus** [comment]
> or do we want to put in the commit file which branch it is in so it only updates from that one?

makes sense

**Comment by thecodacus** [comment]
> image

does this check the commit file based on branch??

**Comment by dustinwloring1988** [comment]
Sorry ment to click reply not edit: Yes will work with any branch as the action will write what branch it is in when it dose the commit.json

**Comment by thecodacus** [comment]
> @thecodacus Sorry ment to click reply not edit: Yes will work with any branch as the action will write what branch it is in when it dose the commit.json.

thats the thing the action will only run on main right ?

**Comment by dustinwloring1988** [comment]
Forgot about that part, now it will run on all.

**Comment by thecodacus** [comment]
> Forgot about that part, now it will run on all.

don't do that. it will be confusing and will be difficult to identify the version

**Comment by dustinwloring1988** [comment]
ok I changed it back, I could think of 2 other ways, what are your thoughts:
1) I could make the branch name a separate action but then that would add even more commits
2) We could change the branch name manually for now when updating the stable branch

**Comment by thecodacus** [comment]
> ok I changed it back, I could think of 2 other ways, what are your thoughts:
> 
> 1. I could make the branch name a separate action but then that would add even more commits
> 2. We could change the branch name manually for now when updating the stable branch

both of these will add commits to stable.

**Comment by dustinwloring1988** [comment]
with doing that the user would have to specify in the setting and we could just default it to stable?

**Comment by thecodacus** [comment]
another option is to check the package.json for version. and check the commit hash of that tag if same then its a stable version

**Comment by dustinwloring1988** [comment]
Updated Now it will auto set then allow the user to change if they so chose.

File: app/lib/hooks/useSettings.tsx [hunk 1]
```diff
@@ -5,19 +5,42 @@ import {
   isLocalModelsEnabled,
   LOCAL_PROVIDERS,
   providersStore,
+  useLatestBranch,
```
Comment (thecodacus) [inline app/lib/hooks/useSettings.tsx:8]: only thing in this entire PR, should not be using  any variable name or function name that start with 'use'  its react convention to name hooks with this syntax. example 'useState', `useSettings`  etc, rename this to `latestBranch`

Comment (dustinwloring1988) [reply app/lib/hooks/useSettings.tsx]: fixed and thank you

**Comment by thecodacus** [comment]
this still had names with `use` infront

**Comment by dustinwloring1988** [comment]
updating now, thought I got them

**Pull request merged**